### DISCUSSION TITLE
Error: FirebaseApp with name {project_number} doesn't exist. Avilable app names: [DEFAULT]

### DISCUSSION THREAD

**Discussion by adriangiulianotucci** [invertase/react-native-firebase#6413] [Q&A]
Hi everyone, I'm trying to update both messaging and app packages. But when my app tries to `firebase.messaging().getToken()` I get this error. In the previous package version that I had installed it didn't happen.

I checked my `google-services.json` and it's ok. It has to be related to the update.

Can someone help me?

Thank you!

EDIT 1: I forgot to mention that only happens in android devices

**Comment by mikehardy** [comment]
From what version to what version? I suspect you crossed a major version boundary that had breaking changes which affected your project, and you may find migration information in the CHANGELOG file?

**Reply by adriangiulianotucci** [reply]
Yes you are right I’m doing a huge upgrade from the following versions to the latests:

    "@react-native-firebase/app": "8.0.0",
    "@react-native-firebase/messaging": "7.2.1"
    
But I’ve read every breaking change and I didn’t find anything. Maybe I can try upgrading in smaller versions…

**Reply by mikehardy** [reply]
My guess is the scoped token API changes? https://github.com/invertase/react-native-firebase/blob/main/CHANGELOG.md#breaking-changes-3

**Reply by adriangiulianotucci** [reply]
I think I've figured it out, I had 2 instances of the same app installed as different apps. I deleted one of them and started working I'm not sure why.

Thanks @mikehardy for your help!

**Reply by mohamadelhinamy** [reply]
Hey @adriangiulianotucci can you elaborate more on what was the issue, I am facing the same issue but cant seem to find any solution?

**Comment by sreenumalae** [comment]
I'm facing the same issue after updating

Initial versions

**"@react-native-firebase/app": "^6.7.1",
 "@react-native-firebase/messaging": "^6.7.1",**

Updated Versions

**"@react-native-firebase/app": "^14.5.0",
"@react-native-firebase/messaging": "^14.5.0",**

when i execute the below code i'm facing issue

```
import messaging from "@react-native-firebase/messaging";

const registerPushNotifications = async () => {
  try {
    await messaging().requestPermission();
  } catch (error) {
    console.debug("Notifications Permission status error:", error);
    return;
  }
  await unregisterPushNotification();
  return messaging().getToken(undefined, "*");
};

const unregisterPushNotification = async () => {
  return messaging().deleteToken(undefined, "*");
};

registerPushNotifications();
```
